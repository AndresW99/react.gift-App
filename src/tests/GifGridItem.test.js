import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';


describe('Testeo del componente GifGridItem', () => {
    
    const title = 'Avengers';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('Debe tomar un SnapShot del componente', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe tener la imagen igual al url y al alt de los props ', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });

    test('debe de tener animate__fadeInDown', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeInDown') ).toBe( true );
        
    })
    
});
