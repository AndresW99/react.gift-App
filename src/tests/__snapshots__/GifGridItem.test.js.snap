// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo del componente GifGridItem Debe tomar un SnapShot del componente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Avengers"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Avengers
     
  </p>
</div>
`;
